File: Server/VideoGameCatalogue.Api/Implementation/Models/VideoGame.cs
namespace VideoGameCatalogue.Api.Implementation.Models;

public sealed class VideoGame
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Genre { get; set; } = string.Empty;

    public string Platform { get; set; } = string.Empty;

    public int ReleaseYear { get; set; }

    public string Developer { get; set; } = string.Empty;

    public string Publisher { get; set; } = string.Empty;

    public int CriticScore { get; set; }

    public string? Summary { get; set; }

    public DateTime UpdatedUtc { get; set; }
}

File: Frontend/src/app/core/models/video-game.ts
export interface VideoGame {
  id: number;
  title: string;
  genre: string;
  platform: string;
  releaseYear: number;
  developer: string;
  publisher: string;
  criticScore: number;
  summary: string | null;
  updatedUtc: string;
}

export interface PagedResponse<TItem> {
  items: TItem[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
}

export interface SaveVideoGameRequest {
  title: string;
  genre: string;
  platform: string;
  releaseYear: number;
  developer: string;
  publisher: string;
  criticScore: number;
  summary: string | null;
}

export interface VideoGameBrowseFilters {
  searchTerm: string;
  genre: string;
  platform: string;
  orderBy: VideoGameOrderBy;
  orderDirection: VideoGameOrderDirection;
}

export type VideoGameOrderBy = 'id' | 'title' | 'releaseYear' | 'criticScore' | 'updatedUtc';

export type VideoGameOrderDirection = 'asc' | 'desc';

export function createEmptyVideoGameBrowseFilters(): VideoGameBrowseFilters {
  return {
    searchTerm: '',
    genre: '',
    platform: '',
    orderBy: 'id',
    orderDirection: 'asc',
  };
}